content/posts/looker-pdt-to-dbt-part-one.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Converting Looker PDTs to dbt models: Reasons why (and why not)"
+date: "2018-12-25"
+author: "Dylan Baker"
+---
+
+Over the past few years, Looker has become a market-leading BI tool. Its data governance and self-service capabilities have now become paramount to thousands of BI stacks. For many, the Third Wave of BI (TM Frank Bien) is well and truly here.
+
+Persistent derived tables (PDTs) are a big part of Looker's success. They allow businesses to quickly transform their data with simple SQL `select` statement, abstracting lots of the data engineering that might otherwise be required. They allow you to easily materialize those models as tables in your data warehouse, speeding up the query times for your end users. 
+
+They are addictive. Build a few PDTs, define an explore. Push. Build a few more. Push. Re-define the calculation behind a dimension. Push. Like most addictions though, there comes a point where you realise it may be detrimental to your health and well-being. Well, that of your BI stack anyway. 
+
+Models can start getting slow, without the ability to optimise fully. Models can get repetitive, with very portions of code written out over and over. Models can break, without the ability to test their output.
+
+## Enter: dbt.
+
+dbt is to PDTs what Reeses is to peanut butter slathered on store-brand chocolate: broadly the same thing, but one far better executed that the other.
+
+From Tristan Handy, the CEO and founder of Fishtown Analytics who build dbt:
+
+> dbt is the T in ELT. It doesn’t extract or load data, but it’s extremely good at transforming data that’s already loaded into your warehouse. This “transform after load” architecture is becoming known as ELT (extract, load, transform).
```

Comment:
I know this is Tristan's quote, but it's a bit weird. It talks about ELT in the first sentence, and then introduces it as a concept later in the paragraph...

Reply:
Maybe move the definition to the first instance of ELT?

Reply:
Or forego/rewrite completely?

Reply:
Have re-written.